pacstall

```diff
@@ -238,6 +239,61 @@ function download() {
 	fi
 }
 
+# source this code block and run like so:
+# 	$ select_options "My message I want to send" "${#array[@]}"
+# This will then output the options given by the user to /tmp/pacstall-select-options, which you can then turn into another array
+function select_options() {
+	rm -f /tmp/pacstall-select-options
+	local message="${1}"
+	local length="${2}"
+
+	if [[ $length -ge 6 ]]; then
+		echo -ne "${message} [${BOLD}1..$length${NC}] [${BIGreen}Y${NC}/${RED}n${NC}] "
+	else
+		echo -ne "${message} [${BOLD}$( seq -s ' ' 1 "$length" )${NC}] [${BIGreen}Y${NC}/${RED}n${NC}] "
+	fi
+	if [[ -z $DISABLE_PROMPTS ]]; then
+		read -r input <&0
+		if [[ $NON_INTERACTIVE ]]; then
+			if [[ -z $input ]]; then
+				echo "Y"
+			fi
+			echo "$input"
+		fi
+	else
+		echo "Y"
+		input="Y"
+	fi
+	if [[ -z $input ]] || [[ $input =~ ^[Yy]$ ]]; then
+		seq -s ' ' 1 "$length" | tee /tmp/pacstall-select-options >/dev/null
+	elif [[ $input =~ ^[Nn]$ ]]; then
+		echo "n" | tee /tmp/pacstall-select-options >/dev/null
+	elif ! [[ $input =~ [a-zA-Z]+ ]] || [[ $input =~ ^[0-9]+$ ]]; then
+		for i in "${input[@]}"; do
+			if [[ $i =~ [0-9]+-[0-9]+ ]] || [[ $i =~ [0-9]+..[0-9]+ ]]; then
+				case "$i" in
+					*-*)
+						split_arr=(${i//-/ })
+						;;
+					*..*)
+						split_arr=(${i//../ })
+						;;
+					*)
+						select_options "$message" "$length"
+						;;
+				esac
+				out+=( $(seq -s ' ' "${split_arr[0]}" "${split_arr[1]}") )
```

Comment:
This should use `mapfile` too, right?

Reply:
This command will have predictable output so I don't think it's needed.

Reply:
I still think the mapfile approach would be cleaner looking though:

```sh
mapfile -O "${#out[@]}" -t out < <(seq -s ' ' "${split_arr[0]}" "${split_arr[1]}")
```

Reply:
> I still think the mapfile approach would be cleaner looking though:
> 
> ```sh
> mapfile -O "${#out[@]}" -t out < <(seq -s ' ' "${split_arr[0]}" "${split_arr[1]}")
> ```

That doesn't seem to look "cleaner" tbh

Reply:
It's just avoiding nested `()` stuff I guess, it won't bother me too much if it's not changed